tests/cbmc/proofs/Makefile.common

```diff
@@ -0,0 +1,279 @@
+GOTO_CC ?= goto-cc
+GOTO_INSTRUMENT ?= goto-instrument
+VIEWER ?= cbmc-viewer
+
+################################################################
+# Setup directory aliases
+SRCDIR ?= $(abspath ../../../..)
+HELPERDIR ?= $(SRCDIR)/tests/cbmc
+PROOFDIR = $(abspath .)
+GOTODIR = $(PROOFDIR)/gotos
+LOGDIR = $(PROOFDIR)/logs
+
+################################################################
+# Useful macros for running commands
+#1: command, 2: flags, 3: log file
+DO_AND_LOG_COMMAND = $(1) $(2) 2>&1 | tee $(3); exit $${PIPESTATUS[0]}
+
+#1: flags, 2: source, 3: target
+DO_GOTO_ANALYZER = $(call DO_AND_LOG_COMMAND,$(GOTO_ANALYZER),$(1) $(2) $(3), $(call LOG_FROM_ENTRY,$(3)))
+
+#1: flags, 2: source, 3: target
+DO_GOTO_CC =  $(call DO_AND_LOG_COMMAND,$(GOTO_CC),$(1) $(2) -o $(3), $(call LOG_FROM_ENTRY,$(3)))
+
+#1: flags, 2: source, 3: target
+DO_GOTO_INSTRUMENT = $(call DO_AND_LOG_COMMAND,$(GOTO_INSTRUMENT),$(1) $(2) $(3), $(call LOG_FROM_ENTRY,$(3)))
+
+#1: message 2: source 3: dest 
+DO_NOOP_COPY = cp $(2) $(3); echo $(1) | tee $(call LOG_FROM_ENTRY,$(3))
+
+################################################################
+# Useful macros translating filenames
+C_FROM_GOTO = $(patsubst $(GOTODIR)%,$(SRCDIR)%,$(1:.goto=.c))
+GOTO_FROM_C = $(patsubst $(SRCDIR)%,$(GOTODIR)%,$(1:.c=.goto))
+LOG_FROM_ENTRY = $(LOGDIR)/$(notdir $(1:.goto=.log))
+LOG_FROM_GOTO = $(patsubst $(GOTODIR)%,$(LOGDIR)%,$(1:.goto=.log))
+
+################################################################
+# Default CBMC flags
+CBMC_VERBOSITY ?= ""
+CBMCFLAGS += --bounds-check
+CBMCFLAGS += --div-by-zero-check
+CBMCFLAGS += --float-overflow-check
+CBMCFLAGS += --nan-check
+CBMCFLAGS += --pointer-check
+CBMCFLAGS += --pointer-overflow-check
+CBMCFLAGS += --signed-overflow-check
+CBMCFLAGS += --undefined-shift-check
+CBMCFLAGS += --unsigned-overflow-check
+CBMCFLAGS += --unwind 1
+CBMCFLAGS += --unwinding-assertions
+
+################################################################
+# Setup the unwindset
+# TODO: CBMC should accept multiple unwindset flags, making the substutions here
+# unnecessary.
+# Tracked in https://github.com/diffblue/cbmc/issues/5185
+ifneq ($(UNWINDSET),)
+CBMC_UNWINDSET := --unwindset $(subst $(SPACE),$(COMMA),$(strip $(UNWINDSET)))
+endif
+CBMCFLAGS += $(CBMC_UNWINDSET)
+
+################################################################
+# Set C compiler defines
+CBMC_OBJECT_BITS ?= 8
+CBMCFLAGS +=  --object-bits $(CBMC_OBJECT_BITS)
+DEFINES += -DCBMC_OBJECT_BITS=$(CBMC_OBJECT_BITS)
+DEFINES += -DCBMC=1
+
+################################################################
+# Setup include directory order
+# HELPERDIR must go before any other includes to override default
+# implementations by stubs
+INC += -I$(HELPERDIR)/include/
+INC += -I$(SRCDIR)
+INC += -I$(SRCDIR)/api
+
+################################################################
+# Enables costly checks (e.g. ones that contain loops)
+# Don't execute deep checks by default
```

Comment:
Please clarify that this is overridden per-proof, because some proofs will need deep checks over a data structure while others only require shallow checks for that very same data structure.

Reply:
Done